### DISCUSSION TITLE
FinTS import fails for "Transfer"

### DISCUSSION THREAD

**Discussion by dklinger** [firefly-iii/firefly-iii#7940] [Firefly III] [answered]
Hi. During FinTS-Import, FireFly returns with error "Die IBAN des Zielkontos wird bereits von einem Bestandskonto oder einer Verbindlichkeit genutzt und kann nicht als Auszahlungsziel verwendet werden." for a specific transaction.
The transaction is a withdrawal from an asset account to a liability account.
I am pretty sure this worked prior to upgrading to 6.0.23. 
Is this a bug? Firefly is correctly identifying the IBAN of the liability account, shouldn't it convert this automatically to a transfer from one account to the other?

**Comment by JC5** [comment]
It depends on the FinTS importer, and I didn't write it. 

https://github.com/bnw/firefly-iii-fints-importer

**Reply by dklinger** [reply]
@JC5 Can you explain why it's the error of the fints-importer?
The error itself is from here and was added 3 months ago: https://github.com/firefly-iii/firefly-iii/blame/67d29b841646b39742fe4a476159a63fdb0424c6/app/Validation/Account/WithdrawalValidation.php#L126

**Reply by dklinger** [reply]
What I mean is: What needs to be changed in fints importer to not have the above code return the error? Is the transaction passed in with incorrect data?

**Reply by dklinger** [reply]
Looks like the problem is, that it's a withdrawal to a liability account. I am not sure why this code was implemented to block this but is there a workaround for this? 
Would it work that I create a rule to convert the withdrawal to a transfer or are the rules executed after the above code?

**Reply by JC5** [reply]
There is no workaround, the code was specifically added to block this from happening. If you want, share with me the original JSON (over email if necessary) so I can run some comparative tests.

The work-around is to either submit the correct transaction type (probably withdrawal in this case), or drop the IBAN. Both things must be implemented by the FinTS importer.

**Reply by dklinger** [reply]
@JC5 I've sent the JSON via mail.

**Comment by dklinger** [comment]
Is there a related issue on GitHub why this is blocked now for liability accounts?
It looks like it does not make sense for my case. As the Bank is submitting the info of the withdrawal with the target account as the IBAN of the liability account, I do not understand why this is an issue of finTS as finTS is just forwarding the info that it gets from the bank.
Scenario:

  *   Asset account with IBAN 1111
  *   Liability account with IBAN 2222
  *   Bank submitting a withdrawal from 1111 to 2222

Why should this not be a supported scenario anymore?

I’m currently driving and can send the json later. Thanks for looking into it.

**Reply by JC5** [reply]
It's perfectly possible to do this. But the account type is wrong. It used to be possible but then it broke installations, so I fixed that :)

**Reply by dklinger** [reply]
Broken installations sound not that cool 😀 
What's the correct way of doing it then
Creating the loan as an asset account?
If so, what are liability accounts supposed for?

**Reply by JC5** [reply]
No the FinTS should use the correct transaction type. I'll see if I can reach out to the developer.

**Reply by dklinger** [reply]
Bringing @bnw into this thread. What are your thoughts about this?

**Reply by JC5** [reply]
Mmm, it looks like it's on me after all. Stay tuned.

**Reply by JC5** [reply, marked as answer]
This will be fixed in the next release of Firefly III. 6.0.24 will feature API 2.0.8 with this fix. I hope to be able to do this over the weekend.

**Reply by dklinger** [reply]
Thanks, 6.0.24 fixed it.
